pkg/apis/componentconfig/types.go

```diff
@@ -572,6 +572,7 @@ type KubeSchedulerConfiguration struct {
 	// HardPodAffinitySymmetricWeight represents the weight of implicit PreferredDuringScheduling affinity rule, in the range 0-100.
 	HardPodAffinitySymmetricWeight int
 	// Indicate the "all topologies" set for empty topologyKey when it's used for PreferredDuringScheduling pod anti-affinity.
+	// DEPRECATED: This is no longer used.
```

Comment:
Agh, this is a mess. Sorry I didn't think of this earlier.

In 1.6 we want to support both the alpha version of this feature and beta version of this feature. (@rrati is working on adding back alpha support, gated by a flag.) In theory, supporting the alpha annotation means continuing to support the --failure-domains flag WHEN THE ALPHA FEATURE IS ENABLED.

However, to move a little closer to deprecating alpha, I think it would be OK to mark the flag deprecated in this PR and remove it from the component config, BUT still interpret empty TopologyKey as meaning  the set api.DefaultFailureDomains WHEN ALPHA FEATURE IS ENABLED. In other words, the default failure domains becomes hard-coded to api.DefaultFailureDomains rather than configurable by command-line flag. Scheduler would continue to support multiple failure domains (indicated by empty TopologyKey) WHEN ALPHA FEATURE IS ENABLED, and we would test it, but the only value it would have is api.DefaultFailureDomains.

Then in a later release we can completely remove support for empty TopologyKey. If we do performance optimizations before then (that assume single topology domain), we would have to gate them on not-alpha.

Does that sound OK?

To answer the question above: I think it would be OK to remove this line (and all support for --failure-domains flag, i.e. all support for configurable FailureDomains).